ephios/plugins/files/models.py

```diff
@@ -31,6 +36,22 @@ def run_on_commit():
     on_commit(run_on_commit, using)
 
 
+def _generate_token():
+    return "".join(
+        random.SystemRandom().choice(string.ascii_uppercase + string.digits) for _ in range(32)
+    )
+
+
+def _token_expiration():
+    return timezone.now() + timedelta(hours=1)
+
+
+class DownloadTicket(models.Model):
+    document = models.ForeignKey(Document, on_delete=models.CASCADE)
+    expiration = models.DateTimeField(default=_token_expiration)
+    token = models.CharField(max_length=255, unique=True, default=_generate_token)
```

Comment:
I thought these would live in the cache in the same [way django-select2 does](https://github.com/codingjoe/django-select2/blob/59798d20c65cd09f6682cc18c50d1d7e2ee4df86/django_select2/views.py#L91-L105) for its heavy widgets :thinking:

Reply:
oh. that's smart.